File: 20주차/이연상/튜플.py
# [PRG] 튜플

def solution(s):
    
    s = s[1:len(s) - 1]
    s = list(s.split("},"))
    s[0] = s[0][1:]
    if(len(s) != 1):
        s[len(s) - 1] = s[len(s) - 1][1:len(s[len(s) - 1]) - 1]
    else:
        s[0] = s[0][0:len(s[0]) - 1]
        
    for i in range(1, len(s) - 1):
        s[i] = s[i][1:]
    
    for i in range(len(s)):
        if(len(s[i]) > 1):
            s[i] = list(map(int, s[i].split(",")))
        else:
            s[i] = [int(s[i])]
    
    s.sort(key=lambda x:len(x))
    result = []
    for i in s:
        for j in i:
            if(j in result):
                continue
            else:
                result.append(j)
    
    
    return result

File: 44주차/이연상/튜플.java
// [PRG] 튜플

import java.util.*;

class Solution {
    public ArrayList<Integer> solution(String s) {
        
        String[] temp = s.split("\\},\\{");
        
        temp[0] = temp[0].substring(2, temp[0].length());
        temp[temp.length - 1] = temp[temp.length - 1].substring(0, temp[temp.length - 1].length() -2);
        ArrayList<Integer>[] listArr = new ArrayList[temp.length];
        
        for(int i = 0; i < temp.length; i++) {
            String[] c = temp[i].split(",");
            
            ArrayList<Integer> list = new ArrayList<>();
            for(int j = 0; j < c.length; j++) {
                list.add(Integer.parseInt(c[j]));
            }
            listArr[i] = list;
        }
        
        Arrays.sort(listArr, (o1, o2) -> o1.size() - o2.size());
        ArrayList<Integer> answer = new ArrayList<>();
        for(ArrayList<Integer> list : listArr) {
            for(int a : list) {
                if(answer.contains(a)) {
                    continue;
                }
                answer.add(a);
            }
        }
        
        return answer;
    }
}